packer_hello_world_example_test.go fails on Apple Silicon

The "Hello World" Packer example and matching test "packer_hello_world_example_test.go" fail if the host plattform is not _linux/amd64_. The cause is easy to determine - the example does use the image `gruntwork/ubuntu-test:16.04`, which is only available for amd64-platforms. Such running the test will first result in the warning `WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested` before the expected text "Hello, World!" is returned, resulting in a failed test:

```
TestPackerHelloWorldExample 2023-06-17T08:59:45Z logger.go:66: Running command docker with args [run gruntwork/packer-hello-world-example cat /test.txt]
TestPackerHelloWorldExample 2023-06-17T08:59:45Z logger.go:66: WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
TestPackerHelloWorldExample 2023-06-17T08:59:45Z logger.go:66: Hello, World!
    packer_hello_world_example_test.go:23: 
                Error Trace:    /workspaces/ws-cdktf-python/terratest/test/packer_hello_world_example_test.go:23
                Error:          Not equal: 
                                expected: "Hello, World!"
                                actual  : "WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested\nHello, World!"
                            
                                Diff:
                                --- Expected
                                +++ Actual
                                @@ -1 +1,2 @@
                                +WARNING: The requested image's platform (linux/amd64) does not match the detected host platform (linux/arm64/v8) and no specific platform was requested
                                 Hello, World!
                Test:           TestPackerHelloWorldExample
--- FAIL: TestPackerHelloWorldExample (5.09s)
FAIL
exit status 1
FAIL    github.com/gruntwork-io/terraform-basic-example 5.110s
```

Leaving aside the question if the example should be updated to fix the problem - how would that problem be handled in a real world scenario? Exchange the base image for an image supporting multi-platform builds? Could the test be updated in a way to except the warning and such be succefull both on amd64 and arm64 hosts? Could we somehow suppress the warning? Any other advise?

--- a/examples/packer-hello-world-example/build.pkr.hcl
+++ b/examples/packer-hello-world-example/build.pkr.hcl
@@ -9,8 +9,10 @@ packer {
 
 source "docker" "ubuntu-docker" {
   changes = ["ENTRYPOINT [\"\"]"]
-  commit  = true
-  image   = "gruntwork/ubuntu-test:16.04"
+
+  commit   = true
+  image    = "gruntwork/ubuntu-test:16.04"
+  platform = "linux/amd64"
 }
 
 build {
@@ -22,6 +24,6 @@ build {
 
   post-processor "docker-tag" {
     repository = "gruntwork/packer-hello-world-example"
-    tag        = ["latest"]
+    tag = ["latest"]
   }
 }

--- a/modules/docker/run.go
+++ b/modules/docker/run.go
@@ -28,6 +28,9 @@ type RunOptions struct {
 	// Assign a name to the container
 	Name string
 
+	// Set platform
+	Platform string
+
 	// If set to true, pass the --privileged flag to 'docker run' to give extended privileges to the container
 	Privileged bool
 
@@ -127,6 +130,10 @@ func formatDockerRunArgs(image string, options *RunOptions) ([]string, error) {
 
 	if options.Name != "" {
 		args = append(args, "--name", options.Name)
+	}
+
+	if options.Platform != "" {
+		args = append(args, "--platform", options.Platform)
 	}
 
 	if options.Privileged {

--- a/test/packer_hello_world_example_test.go
+++ b/test/packer_hello_world_example_test.go
@@ -18,7 +18,11 @@ func TestPackerHelloWorldExample(t *testing.T) {
 	packer.BuildArtifact(t, packerOptions)
 
 	// website::tag::3:: Run the Docker image, read the text file from it, and make sure it contains the expected output.
-	opts := &docker.RunOptions{Command: []string{"cat", "/test.txt"}}
+	opts := &docker.RunOptions{
+		Command:  []string{"cat", "/test.txt"},
+		Platform: "linux/amd64",
+	}
+
 	output := docker.Run(t, "gruntwork/packer-hello-world-example", opts)
 	assert.Equal(t, "Hello, World!", output)
 }

